python/ray/serve/multiplex.py

```diff
@@ -134,7 +134,7 @@ def _push_model_ids_info(self):
         try:
             self.num_models_gauge.set(len(self.models))
 
-            for model_id in self.models:
+            for model_id in list(self.models.keys()):
```

Comment:
Ah this was from back when I tried to fix the threading race condition in https://github.com/ray-project/ray/issues/43106 by making this method thread-safe 😅. I then changed to the alternative approach of making metrics pusher run tasks asynchronously instead of on a thread. This is no longer needed, I will revert this.